milabench/cli.py

```diff
@@ -155,3 +160,73 @@ def install():
 
         mp = _get_multipack(dev=dev)
         mp.do_install(dash=simple_dash, force=force, sync=sync)
+
+    def container():
+        # The container type to create
+        type: Option & str = None
+
+        # Include the dataset in the image
+        include_data: Option & bool = False
+
+        # The tag for the generated container
+        tag: Option & str = "milabench"
+
+        mp = _get_multipack(dev=False)
+
+        if type not in ["docker", "singularity"]:
+            raise ValueError(f"Unsupported type {type}")
+
+        with tempfile.TemporaryDirectory() as base:
+            root = XPath(base)
+
+            # To build containers all your files and directories must be
+            # relative to config_base.
+            config = next(iter(mp.packs.values())).config
+            config_base = XPath(config["config_base"])
+            config_file = XPath(config["config_file"])
+
+            shutil.copytree(config_base, root, dirs_exist_ok=True)
+            config_file.copy(root / "bench.yaml")
+
+            if type == "docker":
+                with (root / "Dockerfile").open("w") as f:
+                    f.write(
+                        dockerfile_template(
+                            milabench_req="git+https://github.com/mila-iqia/milabench.git@container",
+                            include_data=include_data,
+                        )
+                    )
+                subprocess.check_call(["docker", "build", ".", "-t", tag], cwd=root)
```

Comment:
Could we have an option (e.g. `-o DIR`) to output the `root` directory without building the image? That would allow people to build the image themselves without having to install milabench first.

Reply:
This is all done in a temporary directory that is deleted when the process exits, so printing the path out is not really useful.

It would be possible to copy the contents of that directory somewhere else so that the image could be built independently.

Reply:
By "output the root directory" I mean outputting the actual directory, not printing its path. I.e. writing that root directory into the DIR specified with -o.

Reply:
Ok, I've done that.